### ISSUE TITLE
Magic literals calrification

### ISSUE THREAD

**Issue by Ko-Khan** [nus-cs2103-AY2324S2/forum#361]
I was just wondering whether it is necessary to declare a variable for every string printed?

For example,
`public String greet() {
        return "Hi! This is Bit!\nWhat shall we do today?";
    }`
Would this be acceptable as it is pretty self-explanatory?

**Comment by damithc** [comment]
Good question @Ko-Khan 
More generally, which literals should be considered _magic literals_? Let's wait to see what others think.

**Comment by fy17ohhh** [comment]
I think in this case it is descriptive and not haphazard, so both are fine. 
However, for my implementation I still used named constants because I may want to change the content of user feedback messages as I give my bot more personality in the future, and in this case the messages may not be that descriptive? Also I feel that it makes my code more "organized" especially when the messages get longer and I'd rather prefer defining it beforehand, and use a short constant name within each method.

**Comment by zengzihui** [comment]
I feel in this case it should be acceptable, as the function name `greet()` kind of explains what this function is doing.

**Comment by taeewonnn** [comment]
I feel that it is okay to not declare any constant for it here as the method name is well explained. However, I think there is no harm in declaring a constant for better readability, and convenience. The textbook did mention that magic literals should be avoided in general as well
`Avoid any magic literals in general, not just magic numbers.`

**Comment by chiralcentre** [comment]
I think it might be better to use a constant for the greeting message, as it is possible that you may change the greeting message later

**Comment by Ko-Khan** [comment]
Thanks for the advice! I guess magic literals are literals whose meanings are not immediately obvious but there are still advantages to using constants instead.

**Comment by damithc** [comment]
> I guess magic literals are literals whose meanings are not immediately obvious

@Ko-Khan Yes, that's a good observation. So, not every literal needs to be defined as named constants. In some cases the meaning/rationale is obvious from the literal itself.

> but there are still advantages to using constants instead.

That's right, for example, if the same literal is used in multiple places with the same meaning (i.e., they all refer to the same thing), in which case it is better to define it as a constant in one place and  refer to it from other places.

The bottom line: decide case-by-case, considering the above two factors.
